Limit i2c buffer

winkj opened the pull request:
allow limited functionality on platforms with 32 byte I2C buffer sizes; autodetect on platforms we know will have this problem (and are not AVR)

abrauchli commented on sensirion_arch_config.h:
```diff
@@ -35,6 +35,20 @@
+    defined(ARDUINO_ESP8266_RELEASE_2_4_0) || \
+    defined(ARDUINO_ESP8266_RELEASE_2_4_1) || \
+    defined(ARDUINO_ESP8266_RELEASE_2_4_2)
+#define SPS30_USE_I2C_32_BYTE_LIMIT
```
IMHO the wording is a bit unfortunate. I associate `use` more with a feature and limit with an anti-feature.. maybe `#define LIMITED_I2C_BUFFER_SIZE` ?

winkj replied:
Agreed, that sounds better

abrauchli commented on sps30.cpp:
```diff
@@ -107,11 +108,21 @@ s16 sps30_read_data_ready(u16 *data_ready) {
+        u16 value_u16[2];
+        u32 u;
+        f32 f;
+    } val, data[4];
```
The only difference seems `data[4]` vs `data[10]` - use a `#define` instead of duplicating the union definition?

winkj replied:
Will do

winkj commented:
Updated branch addresses
1) Sample sketch handles limited buffer size, prints warning, doesn't print values we can't access
2) renamed PP constant
3) doesn't duplicate union
